api/src/lib.rs

```diff
@@ -1180,17 +1182,17 @@ pub extern "system" fn Java_com_horizen_merkletreenative_InMemoryOptimizedMerkle
         read_raw_pointer(t.j().unwrap() as *const GingerMHT)
     };
 
-    let tree_copy = finalize_ginger_mht(tree);
-
-    return_jobject(&_env, tree_copy, "com/horizen/merkletreenative/InMemoryOptimizedMerkleTree")
-        .into_inner()
+    match finalize_ginger_mht(tree) {
+        Ok(tree_copy) => return_jobject(&_env, tree_copy, "com/horizen/merkletreenative/InMemoryOptimizedMerkleTree").into_inner(),
+        Err(_) => return std::ptr::null::<jobject>() as jobject //CRYPTO_ERROR
```

Comment:
for now is fine. Later when have time, we need to throw a java exception instead of returning the `null`. Also annotate the corresponding java method with `throws ...`
The issue is already present in a backlog.